### ISSUE TITLE
Python test runner umbrella issue

### ISSUE THREAD

**Issue by ojeytonwilliams** [freeCodeCamp/freeCodeCamp#52536]
After https://github.com/freeCodeCamp/freeCodeCamp/pull/52526/ goes in, the following issues will need addressing:

MVP
- [ ] UI: add buttons to run and cancel the learner's code
- [ ] Make sure the service worker isn't cached. i.e. update https://github.com/freeCodeCamp/client-config and deploy a new version
- [ ] Make sure xterm.css _is_ cached and that different versions have different filenames.
- [ ] Make sure the service worker, client/static/python-input-sw.js, updates if we need to change it.
- [ ] Convert e.data

Nice to have

- [ ] Support asyncio, if possible. Pyodide does have a custom event loop (documented [here](https://pyodide.org/en/stable/usage/api/python-api/webloop.html#pyodide.webloop.WebLoop) and introduced in https://github.com/pyodide/pyodide/pull/1158/ ) which is limited, but may be sufficient for our needs.
- [ ] Host pyodide's source ourselves i.e. add the files to the cdn and reference it with [indexURL](https://pyodide.org/en/stable/usage/api/js-api.html#globalThis.loadPyodide)
- [ ] Support testing of print calls and/or give tests access to stdout.
- [ ] Proxy errors to the console. We don't need to redirect stdout, since the terminal shows that, but errors do need to be shown ref. https://github.com/freeCodeCamp/freeCodeCamp/issues/52411
- [ ] If we're running tests in parallel, update the service worker to handle multiple concurrent input requests

Refactors
- [ ] Put code shared between test-evaluator and python-test-evaluator into a common module
- [ ] Rather than using `createTerminal` to open and prepare term, use an async constructor to instantiate an object

**Comment by ShaunSHamilton** [comment]
> UI: add buttons to run and cancel the learner's code

Instead of this, I still think having Campers type out `python3 main.py` in the terminal/preview/console thing is better.

> Proxy errors to the console. We don't need to redirect stdout, since the terminal shows that, but errors do need to be shown ref. https://github.com/freeCodeCamp/freeCodeCamp/issues/52411

Personally, I have always disliked having a "Shell" and a "Console" (e.g. Replit). I would prefer we not do the same, and, instead have a single terminal we:
1) Allow campers to type in
2) Show the output of running the code in

I do not see benefit from further separating the experience to an actual dev/local environment.

**Comment by gikf** [comment]
I don't know where exactly put it on the list, so I'm linking here another python test runner issue: #52448

**Comment by ojeytonwilliams** [comment]
This has got quite stale. Any problems with the python test runner should now be raised as separate issues.
